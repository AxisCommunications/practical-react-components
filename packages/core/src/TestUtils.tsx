import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
