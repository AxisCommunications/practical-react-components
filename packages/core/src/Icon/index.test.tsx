import React from 'react'
import 'jest-styled-components'

import { DeviceIcon } from 'practical-react-components-icons'

import { Icon } from '.'
import { TestRender } from '../TestUtils'

describe('Icons', () => {
  test('Sizes', () => {
    const tree1 = TestRender(<Icon icon={DeviceIcon} size="small" />)
    expect(tree1).toMatchSnapshot()

    const tree2 = TestRender(<Icon icon={DeviceIcon} />)
    expect(tree2).toMatchSnapshot()

    const tree3 = TestRender(<Icon icon={DeviceIcon} size="medium" />)
    expect(tree3).toMatchSnapshot()

    const tree4 = TestRender(<Icon icon={DeviceIcon} size="large" />)
    expect(tree4).toMatchSnapshot()

    const tree5 = TestRender(<Icon icon={DeviceIcon} size="extraLarge" />)
    expect(tree5).toMatchSnapshot()
  })
})
