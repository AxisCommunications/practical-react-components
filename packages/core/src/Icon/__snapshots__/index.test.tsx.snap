// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons Sizes 1`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 16px;
  width: 16px;
}

.c2 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <span
        className="c2"
        role="img"
        size="small"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8v8H5V8h10m1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-2"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Icons Sizes 2`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c2 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-3"
    >
      <span
        className="c2"
        role="img"
        size="medium"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8v8H5V8h10m1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-4"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Icons Sizes 3`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 24px;
  width: 24px;
}

.c2 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-5"
    >
      <span
        className="c2"
        role="img"
        size="medium"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8v8H5V8h10m1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-6"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Icons Sizes 4`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 40px;
  width: 40px;
}

.c2 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-7"
    >
      <span
        className="c2"
        role="img"
        size="large"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8v8H5V8h10m1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-8"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;

exports[`Icons Sizes 5`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  pointer-events: none;
  overflow: hidden;
  vertical-align: middle;
  height: 80px;
  width: 80px;
}

.c2 > svg {
  fill: currentColor;
  height: 100%;
  width: auto;
}

.c4 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c3 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-9"
    >
      <span
        className="c2"
        role="img"
        size="extraLarge"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 8v8H5V8h10m1-2H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4V7c0-.55-.45-1-1-1z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div>
    <div
      className="c3"
      id="layer-10"
    >
      <div
        className="c4"
      />
    </div>
  </div>
</div>
`;
